docs/fsharp/language-reference/plaintext-formatting.md

```diff
@@ -0,0 +1,435 @@
+---
+title: "Plain Text Formatting"
+description: Learn how to use printf and other plain text formatting in F# applications and scripts.
+ms.date: 07/22/2020
+---
+
+# Plain Text Formatting
+
+F# supports type-checked formatting of plain text using `printf`, `printfn`, `sprintf` and related functions.
+For example,
+
+```console
+dotnet fsi
+
+> printfn "Hello %s, %d + %d is %d" "world" 2 2 (2+2);;
+```
+
+gives the output
+
+```fsharp
+Hello world, 2 + 2 is 4
+```
+
+F# also allows structured values to be formatted as plain text.
+For example, consider the following and note how the output has been formatted as a matrix-like display of tuples.
+
+```console
+dotnet fsi
+
+> printfn "%A" [ for i in 1 .. 5 -> [ for j in 1 .. 5 -> (i, j) ] ];;
+
+[[(1, 1); (1, 2); (1, 3); (1, 4); (1, 5)];
+ [(2, 1); (2, 2); (2, 3); (2, 4); (2, 5)];
+ [(3, 1); (3, 2); (3, 3); (3, 4); (3, 5)];
+ [(4, 1); (4, 2); (4, 3); (4, 4); (4, 5)];
+ [(5, 1); (5, 2); (5, 3); (5, 4); (5, 5)]]
+```
+
+Structured plain text formatting is activated when you use the `%A` format in `printf` formatting strings.
+and also when formatting the output of values in F# interactive, where the output includes extra information and is additionally customizable.
+Plain text formatting is also observable through any calls to `x.ToString()` on F# union and record values, including those
+that occur implicitly in debugging, logging and other tooling.
+
+## Checking of `printf`-format strings
```

Comment:
```suggestion
## Check `printf`-format strings
```

Reply:
This isn't quite right.  It's describing how the F# compiler checks format strings at compile-time.  It's not something the user does.

Reply:
Please suggest another change here if you have one?